kingfisher_scrapy/spiders/kenya_makueni.py

```diff
@@ -18,14 +17,19 @@ class KenyaMakueni(IndexSpider):
     # SimpleSpider
     data_type = 'release_package'
 
-    # IndexSpider
-    count_pointer = ''
-    limit = 1000  # > 1000 causes "must be between 1 and 1000"
-    use_page = True
-    start_page = 0
-    param_page = 'pageNumber'
-    base_url = f'https://opencontracting.makueni.go.ke/api/ocds/package/all?pageSize={limit}'
-
     def start_requests(self):
-        url = f'https://opencontracting.makueni.go.ke/api/ocds/release/count?pageSize=1000'
-        yield scrapy.Request(url, meta={'file_name': 'count.json'}, callback=self.parse_list)
+        yield from self.request_page()
+
+    @handle_http_error
+    def parse(self, response):
+        yield from super().parse(response)
+
+        if len(response.json()) > 0:
+            page = response.request.meta['page']
+            yield from self.request_page(page + 1)
+
+    def request_page(self, page=0):
+        step = 1000
+        url = 'https://opencontracting.makueni.go.ke/api/ocds/package/all?pageSize={0}&pageNumber={1}'
+
+        yield self.build_request(url.format(step, page), parameters('pageNumber'), meta={'page': page})
```

Comment:
Construct the URL in fewer steps:

```suggestion
        url = f'https://opencontracting.makueni.go.ke/api/ocds/package/all?pageSize=1000&pageNumber={page}'
        yield self.build_request(url, parameters('pageNumber'), meta={'page': page})
```

Reply:
Done